### PR TITLE
set default tag to package-2021-03-only

### PR DESCRIPTION
**Pull request by vutran01**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>
Updating default tag to build SDK

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit (including refactoring) and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by zhenglaizhang** [comment]
@vutran01 now the default branch is main, we should send PR to main. I updated the PR base branch as main

**Comment by akning-ms** [comment]
Hi @vutran01, which language SDK you are going to release regarding to this default tag change?

**Comment by vutran01** [comment]
Customers have been asking for Go, CLI and terraform support. I assume we'll need Python, Go, JS, and Java

**Comment by akning-ms** [comment]
Reassign to @ArcturusZhang  is better person to review this PR

**Comment by ArcturusZhang** [comment]
I do not think it is a good idea that the default tag of an RP is a "only" tag. This will cause the SDK that are using default tag to generate incomplete SDK for this RP.
Could you introduce a composite tag that consists of all functionalities in this RP to make sure the generated SDK can properly support the APIs in this RP? Like `package-preview-2020-06`

**Comment by vutran01** [comment]
Microsoft.Insights namespace has many RP and owned by many teams, and some might now want SDK generated. Is tag 'package-preview-2020-06' RP that generated SDK before I should include those?

File: specification/applicationinsights/resource-manager/readme.md [hunk 1]
```diff
@@ -222,6 +222,17 @@ directive:
...
 
+### Tag: package-2021-07-only
+
+These settings apply only when `--tag=package-2021-07-only` is specified on the command line.
+
+``` yaml $(tag) == 'package-2021-07-only'
+input-file:
+  - Microsoft.Insights/stable/2021-03-08/workbooks_API.json
+  - Microsoft.Insights/stable/2021-03-08/workbookOperations_API.json
+  - Microsoft.Insights/stable/2020-11-20/workbookTemplates_API.json
```
Comment (msyyc) [inline specification/applicationinsights/resource-manager/readme.md:233]: They are in different folders and I advise to use same folder like : `package-2021-03-only`
image

Comment (vutran01) [reply specification/applicationinsights/resource-manager/readme.md:244]: added a new tag=package-preview-2021-07 and copied all the RP from package-preview-2020-06 and updated the new workbooks paths

**Comment by ArcturusZhang** [comment]
@qiaozha could you please take a look at the JS generation failure here?

**Comment by qiaozha** [comment]
@vutran01  Hi, in your https://github.com/Azure/azure-rest-api-specs/blob/main/specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-03-08/workbooks_API.json file you have refer to different common-type folder such as line 340 refer to v1 folder but line 516 refer to v2 folder, which contains the same definition but with a slightly different in the description. Besides, you have defined your own "Resource" https://github.com/Azure/azure-rest-api-specs/blob/main/specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-03-08/workbooks_API.json#L420 which is also duplicate with those in common-types. You need to change them point to 
 common-types/resource-management/v2/types.json and make use of "allOf" https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/creating-swagger.md#model-inheritance

**Comment by ArcturusZhang** [comment]
Hey @vutran01 any update?

**Review by tonykslee — APPROVED** [review]

File: specification/applicationinsights/resource-manager/readme.md [hunk 2]
```diff
@@ -222,6 +222,31 @@ directive:
   
 ```
 
+### Tag: package-preview-2021-07
+
+These settings apply only when `--tag=package-preview-2021-07` is specified on the command line.
+
+``` yaml $(tag) == 'package-preview-2021-07'
+input-file:
+  - Microsoft.Insights/stable/2015-05-01/aiOperations_API.json
```
[the remark this answers is not in the document]
Comment (vutran01) [reply specification/applicationinsights/resource-manager/readme.md]: removed aioperations

**Comment by vutran01** [comment]
will create new pr with new api-versions for the changes

**Pull request closed without merging**